packages/story-editor/src/components/colorPicker/basicColorList.js

```diff
@@ -135,19 +135,25 @@ function BasicColorList({
               firstIndex++;
               tabIndex = -1;
             }
-            return (
+            const renderedSwatch = (
+              <StyledSwatch
+                key={tooltip ? null : patternAsBackground}
+                onClick={() => handleClick(pattern, isLocal)}
+                pattern={pattern}
+                isSelected={isSelected}
+                isDisabled={isDisabled}
+                tabIndex={tabIndex}
+                title={title}
+              >
+                {isEditMode && <Icons.Cross />}
+              </StyledSwatch>
+            );
+            return tooltip ? (
               <Tooltip key={patternAsBackground} title={tooltip}>
-                <StyledSwatch
-                  onClick={() => handleClick(pattern, isLocal)}
-                  pattern={pattern}
-                  isSelected={isSelected}
-                  isDisabled={isDisabled}
-                  tabIndex={tabIndex}
-                  title={title}
-                >
-                  {isEditMode && <Icons.Cross />}
-                </StyledSwatch>
+                {renderedSwatch}
               </Tooltip>
+            ) : (
+              renderedSwatch
```

Comment:
I think I would prefer something like a <MaybeTooltip /> component in between - so this would just be:

```js
          return (
            <MaybeTooltip key={patternAsBackground} tooltip={tooltip}>
              <StyledSwatch
                onClick={() => handleClick(pattern, isLocal)}
                pattern={pattern}
                isSelected={isSelected}
                isDisabled={isDisabled}
                tabIndex={tabIndex}
                title={title}
              >
                {isEditMode && <Icons.Cross />}
              </StyledSwatch>
            </MaybeTooltip>
          );
```

And then `MaybeTooltip` would be:

```js
function MaybeTooltip({ tooltip, children }) {
  if (!tooltip) return children;
  return <Tooltip title={tooltip}>{children}</Tooltip>
}
```

It simplifies the code a bit and also makes the `key` a bit simpler as it only goes one place.

Naming is not great though - please be more creative than me 😄 .

Alternatively, this could be baked into the existing tooltip component or be made a new component alongside the tooltip but still inside the design system. Maybe an `OptionalTooltip` with a `hasTooltip` property (so it doesn't generally directly depend on the truthiness of the `tooltip` property).

This might be useful in other places in the codebase as well, as I seem to remember similar optional tooltip rendering elsewhere.

Reply:
Sure! I actually tried out that version as well (with `ConditionalTooltip` as the name) but somehow liked the current one more. I'm 50-50 though so if you prefer this one, let's go with that!